lib/daru/io/exporters/excel.rb

```diff
@@ -23,34 +39,96 @@ class Excel < Base
+        #
+        #   #=> #<Daru::DataFrame(2x2)>
+        #   #             x   y
+        #   #  a  b   c   1   3
+        #   #  d  e   f   2   4
+        #
+        #   Daru::IO::Exporters::Excel.new(df,
+        #     "dataframe_df.xls",
+        #     formatting: {
+        #       header: { color: :red, weight: :bold },
+        #       index:  { color: :green },
+        #       data:   { color: :blue }
+        #     }
+        #   ).call
         def initialize(dataframe, path, **options)
           optional_gem 'spreadsheet', '~> 1.1.1'
 
           super(dataframe)
-          @path      = path
-          @options   = options
+          @path       = path
+          @display    = options[:display]    || {}
+          @formatting = options[:formatting] || {}
+
+          @display    = {header: true, data: true, index: true}.merge(@display)
         end
 
-        # @note
-        #
-        #   The +format+ variable used in this method, has to be given
-        #   as options by the user via the +options+ hash input.
-        #
-        #   Signed off by @athityakumar on 03/06/2017 at 7:00PM
         def call
-          book  = Spreadsheet::Workbook.new
-          sheet = book.create_worksheet
+          @book       = Spreadsheet::Workbook.new
+          @sheet      = @book.create_worksheet
+          @row_offset = @display[:header] ? 1 : 0
+          @col_offset = process_col_offset
+
+          write_headers if @display[:header]
 
-          format = Spreadsheet::Format.new color: :blue, weight: :bold
+          if @display[:index] || @display[:data]
+            @dataframe.each_row_with_index.with_index do |row_idx, i|
+              row, idx = row_idx
+
+              write_index(idx, i+@row_offset) if @display[:index]
+              write_data(row,  i+@row_offset) if @display[:data]
+            end
+          end
+
+          @book.write(@path)
+        end
 
-          sheet.row(0).concat(@dataframe.vectors.to_a.map(&:to_s)) # Unfreeze strings
-          sheet.row(0).default_format = format
-          @dataframe.each_row_with_index { |row, i| sheet.row(i+1).concat(row.to_a) }
+        private
+
+        def process_col_offset
+          return 0 unless @display[:index]
+          return 1 unless @dataframe.index.first.is_a?(Array)
+          @dataframe.index.first.count
+        end
+
+        def write_headers
+          @sheet.row(0).default_format = Spreadsheet::Format.new(@formatting[:header] || {})
+          @sheet.row(0).concat([' '] * @col_offset + @dataframe.vectors.to_a.map(&:to_s))
+        end
+
+        def write_index(idx, i)
+          @col_offset.times do |x|
+            @sheet.row(i).set_format(x, Spreadsheet::Format.new(@formatting[:index] || {}))
+          end
+          @sheet.row(i).concat(idx.is_a?(Array) ? idx.to_a.map(&:to_s) : [idx.to_s])
+        end
 
-          book.write(@path)
+        def write_data(row, i)
+          (@col_offset..(@col_offset + @dataframe.vectors.to_a.size-1)).each do |x|
```

Comment:
1. `vectors.to_a.size`, seriously? More idiomatic way?..
2. Instead of `-1`, you should use just excluding range (`...`). Or you can also do `@dataframe.ncols.times`. Or even `row.size.times`
3. `x` is extremely bad name